### ISSUE TITLE
Convening teams + planning issues for Outreachy/Summer of Code

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#5684]
Hi, all! We have some more guidance coming for folks in our summer programs #5667 -- but I also want to encourage people to begin getting to know one another, and looking at who will be working together this summer. 

Some of you are pretty much on a team together on one project, like @alaxalves and @kaustubh-nair -- although how you share the work needs to be planned out -- and others will be working on parts that link up with each other, like @ananyaarun and @CleverFool77 and @gautamig54, for one example (@ananyaarun's work on maps will be featured on several of the pages on PublicLab.org the others will be working on!)

So, this kind of thing needs some discussion, planning, and coordination. Please consider starting to transfer your proposal tasks into "planning issues" with checklists, as outlined in the "Modularity" blog post on https://publiclab.org/software-outreach -- and people can start to organize tasks among team members in such planning issues. 

Some working in parallel may be necessary and helpful! Someone could work on the front-end, while someone else works on the controllers. Or one person could develop re-usable templates that the other plugs into overall page designs! Our UI teams have some to digest in our [new Style Guide draft](https://publiclab.org/notes/warren/05-07-2019/introducing-a-draft-style-guide-for-public-lab)! 

@CleverFool77 @ananyaarun @IshaGupta18 @namangupta01 @sidntrivedi012 @divyabaid16 @Divy123 @MargaretAN9 @ViditChitkara @aashna27 @gautamig54 @kaustubh-nair @rexagod @alaxalves 

**Mentors** -  I've assigned people more projects, just because I hadn't heard from everyone. I'm happy to change, add, reduce your projects, so please feel free to contact me; we can use the ['reviewers' Gitter room](https://gitter.im/publiclab/reviewers) to coordinate. I'm sorry if you got lots of notifications, i don't mean to imply you /have/ to take on more projects, just wanted to get the ball rolling, and we can readjust now. Thanks! 🙌  

Mentors so far: @jainaman224 @ebarry @sagarpreet-chadha @Harshithpabbati @siaw23 @Souravirus @gauravano @Mridul97 @sashadev-sky @SidharthBansal

**Comment by SidharthBansal** [comment]
Thanks Jeff. Let's plan things out.

**Comment by SidharthBansal** [comment]
Hi @publiclab/soc team, all the participants are requested to open Planning issues like we did last summer.
For one project we can open up one planning issue. Example for `MK Rails Upgrade` we will have one planning issue in the MK repository where @kaustubh-nair  and @alaxalves will coordinate.
We can follow this approach for all the projects.
Like Naman and Gautami for `Notification System` 
Naman and Isha for the `Sensor Data Project`
Vidit and @divyabaid16 for `MK synchronous editing` project
and so on.
We can use the assign feature at the planning issue for assigning all the mentors and all the participants on those issues.
This will make us work highly managable and modular.
In case we will create gitter rooms then there are people who uses other IRC channel which does not provide ability to save the previous messages. This can create problem. So i think best approach will be to create planning issues which we did each year.

**Comment by kaustubh-nair** [comment]
Sounds awesome.
@alaxalves and I will have a discussion over the weekend and put up the schedule and division of work on the planning issue accordingly.

**Comment by SidharthBansal** [comment]
That will be simply awesome!!!
You can discuss and interact at the planning issue. Update the header of the issue as you progress through out the project

**Comment by SidharthBansal** [comment]
@publiclab/soc also, we used the `Milestones`. Milestones are used to acertain the progress of the project. So, please add all your issues/prs/ftos into the milestone so that we can evaluate your progress by just reading the milestones.
You can use `gsoc` or `outreachy` label on the issues and prs.
Don't forget to break the planning issue into individual issues and link them to the main planning issue
Thanks

**Comment by SidharthBansal** [comment]
https://github.com/publiclab/plots2/issues/2676 and https://github.com/publiclab/plots2/issues/2388 can help you folks.

**Comment by CleverFool77** [comment]
Hi @jywarren and @SidharthBansal
I had a question. 
Me and @gautamig54 will be working on UI. 
So Should I open an issue containing all the division of works in checklists where we will look into the matter regarding coordination ?

**Comment by sagarpreet-chadha** [comment]
Hi @CleverFool77 ,
Yes please go ahead . I also did the same during my gsoc :) https://github.com/publiclab/plots2/issues/2671

**Comment by aSquare14** [comment]
May I suggest using the Projects feature in GitHub. Issues directly move
from Todo to done as soon as it is closed, and you can move them to work in
progress too. I think it'll help to keep track of issues more efficiently.
Let me know if you need help setting it up. :)

**Comment by SidharthBansal** [comment]
@aSquare14 thanks for the suggestion. I think that feature requires things to be given by mentors. We need to give rights to create projects to the students if we want them to be listed under projects. @jywarren  what you think is better option
Milestones/Projects?

**Comment by aSquare14** [comment]
If I'm not wrong  I think anyone can create a project as long as they're
added to the org :)

**Comment by SidharthBansal** [comment]
What is difference between milestones and project?
Can you please indicate advantages and disadvantages for both?
Thanks

**Comment by aSquare14** [comment]
@SidharthBansal  Found this great answer on stackoverflow ! Do give it a read https://stackoverflow.com/questions/39591795/what-is-the-difference-relationship-between-github-projects-and-milestones

**Comment by SidharthBansal** [comment]
Awesome!!!
I think projects is a better option.

**Comment by gauravano** [comment]
Yeah, using GitHub projects is a better option. We already have many projects - https://github.com/publiclab/plots2/projects :smiley:.

**Comment by CleverFool77** [comment]
Hi everyone.
So Should I create project under the name Outreachy 2019, In which I'll open various tasks like people section , questions section etc ?

**Comment by gauravano** [comment]
Hi, please use name relevant to project like UI redesign , etc

Gaurav Sachdeva

**Comment by gautamig54** [comment]
@CleverFool77 The idea sounds great. You can open issue saying "UI improvements - Summer 2019" or something like this and we can start the discussions and divide the work for different webpages. 
@namangupta01, we can open a project for "Notification System" and start the discussion and divide the work accordingly.

**Comment by SidharthBansal** [comment]
Go ahead Gautami

**Comment by CleverFool77** [comment]
Hi @gautamig54 and everyone
I just opened the project for UI improvements. Now all the new cards and checklists will be added there.
@gautamig54 I guess we can start off with our checklists.
How do we divide them ?
Should I divide the whole project into sections of people, question, dashboard , tags etc.
And every section into further todos parts? 
It would be great to hear some suggestions regarding the protocol followed in projects as I never used one before.
Thank You.

**Comment by SidharthBansal** [comment]
Best way to do things is without restriction in the way you wish to do.
Once different teams are set up, teams can look in each others team and
find good portions which they can adopt.
This approach will give us a quick start with great future enhancements.

**Comment by namangupta01** [comment]
Ya sure Gautami, Can we discuss after 16 may? I have my exams going on.
Thanks!!

**Comment by SidharthBansal** [comment]
Best Of Luck Naman.

**Comment by SidharthBansal** [comment]
Please link milestone/projects/planning issues here.

**Comment by SidharthBansal** [comment]
https://github.com/publiclab/mapknitter/issues/305

**Comment by SidharthBansal** [comment]
https://github.com/publiclab/mapknitter/issues/593
@divyabaid16 @vidit here is issue for you. You asked me to open for you. Please feel free to make it as per your project requirements.
Best of luck

**Comment by sidntrivedi012** [comment]
@jywarren @SidharthBansal Opened the Planning issue in Spectral-workbench.js for the Spectral Workbench Capture project. Please review. https://github.com/publiclab/spectral-workbench.js/issues/92 

I will be adding more details shortly. I am currently somewhat busy due to my university exams which will be ending on 22nd May. Though, I will try to keep up with the work to be done and going on here. 

Thanks :slightly_smiling_face:

**Comment by SidharthBansal** [comment]
Awesome!!!
Best of luck to you all!!!
I have my exams till 19th(on Sunday too!!!).
My university wants us to study even at weekends.. lol...hahaha

**Comment by jywarren** [comment]
This is looking great, folks! I've been trying to go in and give deeper comments on each project's planning issues every couple days. I just did this here: https://github.com/publiclab/leaflet-environmental-layers/issues/168

Actually i'm noticing that there are a lot of projects this year which will be getting started in a separate repo outside `plots2` but with the intention of re-integrating later. This is cool and it's a great approach. I've started trying to connect folks who are working in this way. I know @IshaGupta18 @namangupta01 have just gone through this process quite quickly! This may be relevant to @Rishabh570 as well, in terms of how to architect a stand-alone JS library. 

Note that folks like @gautamig54 @CleverFool77 don't need to do this - your work will be right in `plots2`. But for those working on satellite libraries or general purpose libraries or like @sidntrivedi012 or @rexagod  on a satellite library to another codebase, you can rely on each other a lot for guidance and ideas and helpful critique on how to structure your projects. 

Awesome to see everyone getting organized and ready! 🎉 🎉 🎉

**Comment by gautamig54** [comment]
@CleverFool77 Thanks for opening a project for UI improvements. I think it is better to have a separate issue for each page which can be added in the to-do list for the project. Also, I will open up an issue which will list down the pages which we need to focus on or any other UI improvements and we can discuss our summer plan there. It will be helpful for an organised start to the project as our entire summer plan will be at a single place.

**Comment by jywarren** [comment]
Gautami this sounds super!

**Comment by gautamig54** [comment]
I have opened a project for the Notification System. @namangupta01 Can you have a look at it and add if I missed something. Also, should I create an issue for the Notification System project for discussions?

**Comment by SidharthBansal** [comment]
We can have discussion in the planning issue itself. At the beginning of
issue, you can have the editable checklist.
Thanks

**Comment by Divy123** [comment]
@aashna27 , @publiclab/mentors I have added the project for Image Sequencer here:
https://github.com/publiclab/image-sequencer/projects/2
 Aashna you can add your set of issues as well by breaking them up as suitable for you. Also we can collaborate upon and keep on breaking some issues as suitable.

**Comment by CleverFool77** [comment]
> @CleverFool77 Thanks for opening a project for UI improvements. I think it is better to have a separate issue for each page which can be added in the to-do list for the project. Also, I will open up an issue which will list down the pages which we need to focus on or any other UI improvements and we can discuss our summer plan there. It will be helpful for an organised start to the project as our entire summer plan will be at a single place.

Hi @gautamig54 .
That's exactly what I followed. I tried keeping different issue for each section.
And In each section , I added further checklists based on division of work.
We'll be adding these issues in todos of projects. It would make the process convenient and easy
Thanks !!!

**Comment by namangupta01** [comment]
Hi @gautamig54, I have added some tasks in the project and will add some soon.

**Comment by SidharthBansal** [comment]
Great

**Comment by aSquare14** [comment]
@SidharthBansal  Can you please add me to the reviewers chat on gitter so that I can discuss a few things with you all ! :)

**Comment by jywarren** [comment]
Just noting this, sorry for slow reply -- the reviewers room is at
https://gitter.im/publiclab/reviewers ! I believe it's public!

**Comment by gauravano** [comment]
@jywarren actually the reviewers' group is an invite-only group. I just added you @aSquare14 to the group.     Welcome :tada: !!

**Comment by aSquare14** [comment]
Thanks.

**Comment by jywarren** [comment]
Great work everyone!!!!
